site24x7/site24x7.go

```diff
@@ -0,0 +1,113 @@
+package site24x7
+
+import (
+	"encoding/json"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/target/goalert/alert"
+	"github.com/target/goalert/integrationkey"
+	"github.com/target/goalert/permission"
+	"github.com/target/goalert/retry"
+	"github.com/target/goalert/util/errutil"
+	"github.com/target/goalert/util/log"
+	"github.com/target/goalert/validation/validate"
+)
+
+type site24x7Post struct {
+	MONITOR_DASHBOARD_LINK  string
+	MONITORTYPE             string
+	STATUS                  string
+	REASON                  string
+	MONITORNAME             string
+	FAILED_LOCATIONS        string
+	INCIDENT_REASON         string
+	OUTAGE_TIME_UNIX_FORMAT string
+	MONITORURL              string
+	MONITOR_GROUPNAME       string
+	INCIDENT_TIME           string
+	INCIDENT_TIME_ISO       string
+	RCA_LINK                string
+	//ct                      int
+}
+
+func clientError(w http.ResponseWriter, code int, err error) bool {
+	if err == nil {
+		return false
+	}
+
+	http.Error(w, http.StatusText(code), code)
+	return true
+}
+
+func Site24x7ToEventsAPI(aDB alert.Store, intDB integrationkey.Store) http.HandlerFunc {
+
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		ctx := r.Context()
+
+		err := permission.LimitCheckAny(ctx, permission.Service)
+		if errutil.HTTPError(ctx, w, err) {
+			return
+		}
+		serviceID := permission.ServiceID(ctx)
+
+		var g site24x7Post
+		err = json.NewDecoder(r.Body).Decode(&g)
+		if clientError(w, http.StatusBadRequest, err) {
+			log.Logf(ctx, "bad request from site24x7: %v", err)
+			return
+		}
+
+		ctx = log.WithFields(ctx, log.Fields{
+			"RuleURL": g.MONITOR_DASHBOARD_LINK,
+			"State":   g.STATUS,
+		})
+
+		var site24x7State alert.Status
+		switch g.STATUS {
+		case "DOWN":
```

Comment:
The error states can be simplified to a single case:
```go
case "DOWN", "CRITICAL", "TROUBLE":
```

Reply:
Thanks for the tip, I've simplified these.